Yes you've typed the function but I'm getting these typing failures:
```
Run mypy --config-file mypy.ini src/build123d
  mypy --config-file mypy.ini src/build123d
  shell: /usr/bin/bash -e {0}
  env:
    UV_PYTHON: 3.10
    VIRTUAL_ENV: /home/runner/work/build123d/build123d/.venv
    UV_CACHE_DIR: /home/runner/work/_temp/setup-uv-cache
src/build123d/topology/two_d.py:108: error: Skipping analyzing "ocp_gordon": module is installed, but missing library stubs or py.typed marker  [import-untyped]
src/build123d/topology/two_d.py:108: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 37 source files)
Error: Process completed with exit code 1.
```
I not 100% sure but I think the problem is a missing: src_py/ocp_gordon/py.typed (an empty file) which will indicate that your API is typed.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -44,7 +44,7 @@ dependencies = [
     "ipython >= 8.0.0, < 10",
     "lib3mf >= 2.4.1",
     "ocpsvg >= 0.5, < 0.6",
-    "ocp_gordon >= 0.1.10",
+    "ocp_gordon >= 0.1.12",
     "trianglesolver",
     "sympy",
     "scipy",

--- a/src/build123d/topology/two_d.py
+++ b/src/build123d/topology/two_d.py
@@ -866,38 +866,6 @@ class Face(Mixin2D, Shape[TopoDS_Face]):
         return Face(TopoDS.Face_s(_extrude_topods_shape(obj.wrapped, direction)))
 
     @classmethod
-    def gordon_surface(
-        cls,
-        profiles: Iterable[Edge],
-        guides: Iterable[Edge],
-        tolerance: float = 3e-4,
-    ) -> Face:
-        """
-        Creates a Gordon surface from a network of profile and guide curves.
-
-        Args:
-            profiles (Iterable[Edge]): Edges representing profile curves.
-            guides (Iterable[Edge]): Edges representing guide curves.
-            tolerance (float, optional): Tolerance for surface creation and
-                intersection calculations.
-
-        Returns:
-            Face: the interpolated Gordon surface
-        """
-        ocp_profiles = [BRep_Tool.Curve_s(edge.wrapped, 0, 1) for edge in profiles]
-        ocp_guides = [BRep_Tool.Curve_s(edge.wrapped, 0, 1) for edge in guides]
-
-        gordon_bspline_surface = interpolate_curve_network(
-            ocp_profiles, ocp_guides, tolerance=tolerance
-        )
-
-        return cls(
-            BRepBuilderAPI_MakeFace(
-                gordon_bspline_surface, Precision.Confusion_s()
-            ).Face()
-        )
-
-    @classmethod
     def make_bezier_surface(
         cls,
         points: list[list[VectorLike]],
@@ -945,6 +913,47 @@ class Face(Mixin2D, Shape[TopoDS_Face]):
             bezier = Geom_BezierSurface(points_)
 
         return cls(BRepBuilderAPI_MakeFace(bezier, Precision.Confusion_s()).Face())
+
+    @classmethod
+    def make_gordon_surface(
+        cls,
+        profiles: Iterable[Edge],
+        guides: Iterable[Edge],
+        tolerance: float = 3e-4,
+    ) -> Face:
+        """
+        Creates a Gordon surface from a network of profile and guide curves.
+
+        Args:
+            profiles (Iterable[Edge]): Edges representing profile curves.
+            guides (Iterable[Edge]): Edges representing guide curves.
+            tolerance (float, optional): Tolerance for surface creation and
+                intersection calculations.
+
+        Raises:
+            ValueError: Input edge cannot be empty
+
+        Returns:
+            Face: the interpolated Gordon surface
+        """
+
+        def to_geom_curve(edge: Edge):
+            if edge.wrapped is None:
+                raise ValueError("input edge cannot be empty")
+            return BRep_Tool.Curve_s(edge.wrapped, 0, 1)
+
+        ocp_profiles = [to_geom_curve(edge) for edge in profiles]
+        ocp_guides = [to_geom_curve(edge) for edge in guides]
+
+        gordon_bspline_surface = interpolate_curve_network(
+            ocp_profiles, ocp_guides, tolerance=tolerance
+        )
+
+        return cls(
+            BRepBuilderAPI_MakeFace(
+                gordon_bspline_surface, Precision.Confusion_s()
+            ).Face()
+        )
 
     @classmethod
     def make_plane(

--- a/tests/test_direct_api/test_face.py
+++ b/tests/test_direct_api/test_face.py
@@ -434,6 +434,22 @@ class TestFace(unittest.TestCase):
             u=1.0, v=1.0, expected_point=profiles[-1].position_at(1.0)
         )
 
+        temp_curve = profiles[0]
+        with self.assertRaises(ValueError):
+            profiles[0] = Edge()
+            tolerance = 3e-4
+            gordon_surface = Face.make_gordon_surface(
+                profiles, guides, tolerance=tolerance
+            )
+
+        profiles[0] = temp_curve
+        with self.assertRaises(ValueError):
+            guides[0] = Edge()
+            tolerance = 3e-4
+            gordon_surface = Face.make_gordon_surface(
+                profiles, guides, tolerance=tolerance
+            )
+
     def test_make_surface(self):
         corners = [Vector(x, y) for x in [-50.5, 50.5] for y in [-24.5, 24.5]]
         net_exterior = Wire(

--- a/tests/test_direct_api/test_face.py
+++ b/tests/test_direct_api/test_face.py
@@ -359,6 +359,81 @@ class TestFace(unittest.TestCase):
         self.assertAlmostEqual(loc.position, (0.0, 1.0, 1.5), 5)
         self.assertAlmostEqual(loc.orientation, (0, -90, 0), 5)
 
+    def test_make_gordon_surface(self):
+        def create_test_curves(
+            num_profiles: int = 3,
+            num_guides: int = 4,
+            u_range: float = 1.0,
+            v_range: float = 1.0,
+        ):
+            profiles: list[Edge] = []
+            guides: list[Edge] = []
+
+            intersection_points = [
+                [(0.0, 0.0, 0.0) for _ in range(num_guides)]
+                for _ in range(num_profiles)
+            ]
+
+            for i in range(num_profiles):
+                for j in range(num_guides):
+                    u = i * u_range / (num_profiles - 1)
+                    v = j * v_range / (num_guides - 1)
+                    z = 0.2 * math.sin(u * math.pi) * math.cos(v * math.pi)
+                    intersection_points[i][j] = (u, v, z)
+
+            for i in range(num_profiles):
+                points = [intersection_points[i][j] for j in range(num_guides)]
+                profiles.append(Spline(points))
+
+            for j in range(num_guides):
+                points = [intersection_points[i][j] for i in range(num_profiles)]
+                guides.append(Spline(points))
+
+            return profiles, guides
+
+        profiles, guides = create_test_curves()
+
+        tolerance = 3e-4
+        gordon_surface = Face.make_gordon_surface(profiles, guides, tolerance=tolerance)
+
+        self.assertIsInstance(
+            gordon_surface, Face, "The returned object should be a Face."
+        )
+
+        def point_at_uv_against_expected(u: float, v: float, expected_point: Vector):
+            point_at_uv = gordon_surface.position_at(u, v)
+            self.assertAlmostEqual(
+                point_at_uv.X,
+                expected_point.X,
+                delta=tolerance,
+                msg=f"X coordinate mismatch at ({u},{v})",
+            )
+            self.assertAlmostEqual(
+                point_at_uv.Y,
+                expected_point.Y,
+                delta=tolerance,
+                msg=f"Y coordinate mismatch at ({u},{v})",
+            )
+            self.assertAlmostEqual(
+                point_at_uv.Z,
+                expected_point.Z,
+                delta=tolerance,
+                msg=f"Z coordinate mismatch at ({u},{v})",
+            )
+
+        point_at_uv_against_expected(
+            u=0.0, v=0.0, expected_point=guides[0].position_at(0.0)
+        )
+        point_at_uv_against_expected(
+            u=1.0, v=0.0, expected_point=profiles[0].position_at(1.0)
+        )
+        point_at_uv_against_expected(
+            u=0.0, v=1.0, expected_point=guides[0].position_at(1.0)
+        )
+        point_at_uv_against_expected(
+            u=1.0, v=1.0, expected_point=profiles[-1].position_at(1.0)
+        )
+
     def test_make_surface(self):
         corners = [Vector(x, y) for x in [-50.5, 50.5] for y in [-24.5, 24.5]]
         net_exterior = Wire(
@@ -501,78 +576,6 @@ class TestFace(unittest.TestCase):
                     Edge.make_line((0, 0), (0, 1)),
                 ]
             )
-
-    def test_gordon_surface(self):
-        def create_test_curves(
-            num_profiles: int = 3,
-            num_guides: int = 4,
-            u_range: float = 1.0,
-            v_range: float = 1.0,
-        ):
-            profiles: list[Edge] = []
-            guides: list[Edge] = []
-
-            intersection_points = [
-                [(0.0, 0.0, 0.0) for _ in range(num_guides)]
-                for _ in range(num_profiles)
-            ]
-
-            for i in range(num_profiles):
-                for j in range(num_guides):
-                    u = i * u_range / (num_profiles - 1)
-                    v = j * v_range / (num_guides - 1)
-                    z = 0.2 * math.sin(u * math.pi) * math.cos(v * math.pi)
-                    intersection_points[i][j] = (u, v, z)
-
-            for i in range(num_profiles):
-                points = [intersection_points[i][j] for j in range(num_guides)]
-                profiles.append(Spline(points))
-
-            for j in range(num_guides):
-                points = [intersection_points[i][j] for i in range(num_profiles)]
-                guides.append(Spline(points))
-
-            return profiles, guides
-
-        profiles, guides = create_test_curves()
-
-        tolerance = 3e-4
-        gordon_surface = Face.gordon_surface(profiles, guides, tolerance=tolerance)
-
-        self.assertIsInstance(
-            gordon_surface, Face, "The returned object should be a Face."
-        )
-
-        def point_at_uv_against_expected(u: float, v: float, expected_point: Vector):
-            point_at_uv = gordon_surface.position_at(u, v)
-            self.assertAlmostEqual(
-                point_at_uv.X,
-                expected_point.X,
-                delta=tolerance,
-                msg=f"X coordinate mismatch at ({u},{v})",
-            )
-            self.assertAlmostEqual(
-                point_at_uv.Y,
-                expected_point.Y,
-                delta=tolerance,
-                msg=f"Y coordinate mismatch at ({u},{v})",
-            )
-            self.assertAlmostEqual(
-                point_at_uv.Z,
-                expected_point.Z,
-                delta=tolerance,
-                msg=f"Z coordinate mismatch at ({u},{v})",
-            )
-
-        point_at_uv_against_expected(
-            u=1.0, v=0.0, expected_point=profiles[0].position_at(1.0)
-        )
-        point_at_uv_against_expected(
-            u=0.0, v=1.0, expected_point=guides[0].position_at(1.0)
-        )
-        point_at_uv_against_expected(
-            u=1.0, v=1.0, expected_point=profiles[-1].position_at(1.0)
-        )
 
     # def test_to_arcs(self):
     #     with BuildSketch() as bs:

--- a/tests/test_direct_api/test_face.py
+++ b/tests/test_direct_api/test_face.py
@@ -435,17 +435,15 @@ class TestFace(unittest.TestCase):
         )
 
         temp_curve = profiles[0]
+        profiles[0] = Edge()
         with self.assertRaises(ValueError):
-            profiles[0] = Edge()
-            tolerance = 3e-4
             gordon_surface = Face.make_gordon_surface(
                 profiles, guides, tolerance=tolerance
             )
 
         profiles[0] = temp_curve
+        guides[0] = Edge()
         with self.assertRaises(ValueError):
-            guides[0] = Edge()
-            tolerance = 3e-4
             gordon_surface = Face.make_gordon_surface(
                 profiles, guides, tolerance=tolerance
             )

